### PR TITLE
Filter data sorting

### PR DESCRIPTION
**Pull request by mxsnyder**
This is an update for the databases page logic to account for a webservice-driven provider, rather than scraping HTML.

### CODE REVIEW

File: js/apps/filter-page/services/filterService.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+			return _.uniq(_.pluck(tagInfoArr, 'Name'), name => name);
+		};
+
+		const getFamilyTags = (familyName, tagInfoArr) => {
+			const familyTagInfo = _.where(tagInfoArr, { Name: familyName });
+			return _.uniq(_.sortBy(_.pluck(familyTagInfo, 'Tag'), tag => tag), tag => tag);
+		};
+
+		const getFamilyType = (familyName, tagInfoArr) => {
+			return _.findWhere(tagInfoArr, { Name: familyName }).Type;
```
Comment (martypowell) [inline js/apps/filter-page/services/filterService.js]: If there are not results, will this throw an error?

File: js/apps/filter-page/controllers/filterPageCtrl.js [hunk 2]
```diff
@@ -84,14 +75,25 @@
...
+			let foundFilterFamily = filterFamily;
+
+			if (isTagInfo) {
+				foundFilterFamily = _.where(self.filters, { name: filter.Name });
+				if (foundFilterFamily.length === 1) {
+					foundFilterFamily = foundFilterFamily[0];
+				}
+			}
+
+			const isPickOne = foundFilterFamily.type.toLowerCase() === 'one';
```
[the remark this answers is not in the document]
Comment (mxsnyder) [reply js/apps/filter-page/controllers/filterPageCtrl.js]: done

File: js/apps/filter-page/directives/tag.js [hunk 3]
```diff
@@ -33,7 +29,7 @@
 		return directive;
 	};
 
-	tagDirective.$inject = ['tagParsingService'];
+	tagDirective.$inject = [];
```
[the remark this answers is not in the document]
Comment (mxsnyder) [reply js/apps/filter-page/directives/tag.js]: done

File: js/apps/filter-page/filters/cardVisibilityFilter.js [hunk 4]
```diff
@@ -21,7 +21,7 @@
 		};
 	};
 
-	cardVisibilityFilter.$inject = ['tagParsingService'];
+	cardVisibilityFilter.$inject = [];
```
[the remark this answers is not in the document]
Comment (mxsnyder) [reply js/apps/filter-page/filters/cardVisibilityFilter.js]: done

File: js/apps/filter-page/services/filterService.js (hunk 1 above)
Comment (mxsnyder) [reply js/apps/filter-page/services/filterService.js]: yep - done

**Comment by mxsnyder** [comment]
Thanks!

**Pull request merged**